src/foam/u2/view/FObjectView.js

```diff
@@ -7,30 +7,38 @@
 foam.CLASS({
   package: 'foam.u2.view',
   name: 'FObjectView',
-  extends: 'foam.u2.Controller',
-
+  extends: 'foam.u2.View',
   documentation: 'View for editing FObjects.',
-
-  // css: '^:read-only { border: none; background: rgba(0,0,0,0); }'
-
   properties: [
     {
       class: 'String',
       name: 'objectClass',
-      displayWidth: 70,
-      view: { class: 'foam.u2.TextField', size: 80 },
+      label: '',
+      visibilityExpression: function(choices) {
+        return choices.length > 1 ?
+          foam.u2.Visibility.RW :
+          foam.u2.Visibility.HIDDEN;
+      },
+      view: function(_, x) {
+        return foam.u2.view.ChoiceView.create({
+          choices: x.data.choices
+        }, x);
+      },
       postSet: function(oldValue, newValue) {
         if ( newValue !== oldValue ) {
           var m = this.__context__.lookup(newValue, true);
           if ( m ) {
-            this.data = m.create(this.data, this);
+            var n = m.create(null, this);
```

Comment:
Why was this unrolled into multiple steps?  Not saying its wrong but curious what the difference was functionally.

Reply:
Doing a copyFrom is quite a bit different from doing it directly in create (which invokes initArgs)